doc/developer/style.md

```diff
@@ -116,6 +116,77 @@ Some key principles to highlight are:
 * Avoid contractions and spell out words in full.
 * Word choices to be mindful of: find vs. exists, may vs. can vs. might.
 
+## Log message style
+
+We use the [`tracing` crate](https://docs.rs/tracing/latest/tracing/)'s log
+macros. Observe the following guidelines:
+
+  * Use sentence fragment style. Say "beginning operation", not
+    "Beginning operation."
+
+  * Prefer structured fields to string formatting:
+
+    ```
+    fn handle_persist_thingy(shard_id: ShardId) {
+        // Preferred.
+        info!(shard_id = ?the_shard, "handling persist thingy");
+        // Acceptable.
+        info!("handling persist thingy for shard {}", shard_id);
+    }
+    ```
+
+Messages at each level must meet the indicated standard:
+
+* **`ERROR`**: Reports an error that certainly indicates data corruption, data
+  loss, unavailability, or an invariant violation that results in undefined
+  behavior. These errors are reported to Sentry and may eventually page the
+  on-call engineer. Use *judiciously*.
```

Comment:
if errors are truly "data corruption, data loss, unavailability, or an invariant violation that results in undefined behavior", why is this "_may eventually_ page"? seems like the language here could be strengthened

Reply:
Good point! Strengthening to "will eventually."